src/core/Akka.Streams/Stage/GraphStage.cs

```diff
@@ -1683,8 +1695,77 @@ protected Action<T> GetAsyncCallback<T>(Action<T> handler)
         /// <param name="handler">TBD</param>
         /// <returns>TBD</returns>
         protected Action GetAsyncCallback(Action handler)
-            => () => Interpreter.OnAsyncInput(this, NotUsed.Instance, _ => handler());
+            => () => Interpreter.OnAsyncInput(this, NotUsed.Instance, NoPromise, _ => handler());
 
+        protected Func<T, Task<Done>> GetAsyncCallbackAsync<T>(Action<T> handler)
```

Comment:
New async callback API, instead of returning an `Action<T>`, this new API returns a `Func<T, Task<Done>>` where external callers can observe the result of any async callback invocations.